### PR TITLE
drivers: can: mcp251xfd: Don't fail silently without forwarding error

### PR DESCRIPTION
**Pull request by talih0**
Continues PR: https://github.com/zephyrproject-rtos/zephyr/pull/73714

### CODE REVIEW

**Comment by talih0** [comment]
I'm continuing the original PR, but there is a compliance error complaining about the name of the author in the first two commits. @henrikbrixandersen, any suggestions?

**Comment by henrikbrixandersen** [comment]
> I'm continuing the original PR, but there is a compliance error complaining about the name of the author in the first two commits. @henrikbrixandersen, any suggestions?

The only suggestion I can give is to fix the compliance error?

**Comment by talih0** [comment]
Sorry, I don't understand your reply.

**Comment by henrikbrixandersen** [comment]
> Sorry, I don't understand your reply.

From the compliance check:

```
Commit 3ba7fa789d:
1: UC2 Signed-off-by: must have a full name

Commit 08c86eeedf:
1: UC2 Signed-off-by: must have a full name
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

 3ba7fa789def879ab66986a21da4f310acbcd28d: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
 08c86eeedf461da1f3c2b3f7c05e5a61d3afcd7e: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
```

The `Signed-off-by` must have a real name consisting of a first name, a last name, and an email address. The commits, you've submitted here, do not meet that criteria.

**Comment by talih0** [comment]
> > Sorry, I don't understand your reply.
> 
> From the compliance check:
> 
> ```
> Commit 3ba7fa789d:
> 1: UC2 Signed-off-by: must have a full name
> 
> Commit 08c86eeedf:
> 1: UC2 Signed-off-by: must have a full name
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
>  3ba7fa789def879ab66986a21da4f310acbcd28d: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
>  08c86eeedf461da1f3c2b3f7c05e5a61d3afcd7e: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
> ```
> 
> The `Signed-off-by` must have a real name consisting of a first name, a last name, and an email address. The commits, you've submitted here, do not meet that criteria.

The first two commits that trigger the compliance problem are from PR: https://github.com/zephyrproject-rtos/zephyr/pull/73709 that was abandoned. I just cherry picked the commits and added my fix on top. I do not know the original author's full name unfortunately.

I guess I'll take a reasonable guess at their name.

**Comment by talih0** [comment]
> > > Sorry, I don't understand your reply.
> > 
> > 
> > From the compliance check:
> > ```
> > Commit 3ba7fa789d:
> > 1: UC2 Signed-off-by: must have a full name
> > 
> > Commit 08c86eeedf:
> > 1: UC2 Signed-off-by: must have a full name
> > Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> > 
> >  3ba7fa789def879ab66986a21da4f310acbcd28d: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
> >  08c86eeedf461da1f3c2b3f7c05e5a61d3afcd7e: author email (melvinvdb <vdbergmelvin@gmail.com>) does not follow the syntax: First Last <email>.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The `Signed-off-by` must have a real name consisting of a first name, a last name, and an email address. The commits, you've submitted here, do not meet that criteria.
> 
> The first two commits that trigger the compliance problem are from PR: #73709 that was abandoned. I just cherry picked the commits and added my fix on top. I do not know the original author's full name unfortunately.
> 
> I guess I'll take a reasonable guess at their name.

On second thoughts, I don't think it's a good to guess the name. I ended up just changing the author name to mine and giving credit to <melvindb> in the commit message.

**Review by str4t0m — COMMENTED** [review]
Looks good.
One NIT: Could you update the second commit message to point to the PR #73714 instead of #73709.

File: drivers/can/can_mcp251xfd.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ BUILD_ASSERT(MCP251XFD_TEF_FIFO_SIZE + MCP251XFD_TX_QUEUE_SIZE +
 #define MCP251XFD_MAX_INT_HANDLER_CALLS  10
 #define MCP251XFD_INT_HANDLER_SLEEP_USEC 10000
 
+#define MCP251XFD_RESET_DELAY_USEC 5000
```
Comment (henrikbrixandersen) [inline drivers/can/can_mcp251xfd.h:72]: Is this value from a datasheet or just a random value that seems to work? Please add a code comment.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please comment on where the 5 msec delay comes from and ensure the Git commit messages follow the established format (e.g. put "Fixes: #..." on a separate line last in the message, include a description of _why_ something was changes, not just _what_ was changed).

File: drivers/can/can_mcp251xfd.h (hunk 1 above)
Comment (henrikbrixandersen) [reply drivers/can/can_mcp251xfd.h:72]: I know you are not the original author, but you are submitting something for inclusion in the project and therefore responsible for the changes being proposed.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**